src/shared/clients/iamClient.ts

```diff
@@ -93,4 +93,11 @@ export class DefaultIamClient {
 
         return policies
     }
+
+    public async getIAMRoleFromInstanceProfile(instanceProfileArn: string): Promise<IAM.Role> {
+        const client = await this.createSdkClient()
+        const instanceProfileName = this.getFriendlyName(instanceProfileArn)
+        const response = await client.getInstanceProfile({ InstanceProfileName: instanceProfileName }).promise()
+        return response.InstanceProfile.Roles[0]
```

Comment:
Could this return undefined? If so, the return type seems wrong. Or maybe we should throw if `Roles.length == 0`

Reply:
I was under the understanding that an instance profile is created to link a role to an instance, so it seems strange it would have no roles. However, to be safe I changed it to throw here.